src/charm.py

```diff
@@ -2392,12 +2495,21 @@ def _api_update_config(self) -> None:
             "max_wal_senders": 25,
             "wal_keep_size": self.config.durability_wal_keep_size,
         }
+
+        # Add restart-required worker process parameters via Patroni API
+        worker_configs = self._calculate_worker_process_config()
+        restart_required_params = ["max_worker_processes", "max_logical_replication_workers"]
+
+        for param in restart_required_params:
+            if param in worker_configs:
+                cfg_patch[param] = worker_configs[param]
```

Comment:
I think we can just set `max_worker_processes` and `max_logical_replication_workers` directly in the `cfg_patch` dict above.